controllers/storagecluster/cephfilesystem.go

```diff
@@ -51,14 +52,56 @@ func (r *StorageClusterReconciler) newCephFilesystemInstances(initStorageCluster
 		// standalone deployment that isn't in provider cluster will not
 		// have storageProfile, we need to define default dataPool, if
 		// storageProfile is set this will be overridden.
-		ret.Spec.DataPools = []cephv1.NamedPoolSpec{
-			{
-				PoolSpec: cephv1.PoolSpec{
-					DeviceClass:   generateDeviceClass(initStorageCluster),
-					Replicated:    generateCephReplicatedSpec(initStorageCluster, "data"),
-					FailureDomain: initStorageCluster.Status.FailureDomain,
+		dataPool := initStorageCluster.Spec.ManagedResources.CephFilesystems.DataPool
+		// Use the specified data pool if it is set, otherwise use the default pool
+		if reflect.DeepEqual(dataPool, cephv1.NamedPoolSpec{}) {
```

Comment:
if this check is passed dataPool is empty but you are trying to use it to filesystem creation, please check you might need to add '!' to it.

Reply:
Thanks for pointing out, there were some mistakes with the equality checks, fixed now.